enterprise/api/v1/serializers.py

```diff
@@ -954,17 +954,35 @@ def validate(self, data):  # pylint: disable=arguments-differ
         return data
 
 
+class LicensesInfoSerializer(serializers.Serializer):
+    """
+    Nested serializer class to allow for many license info dictionaries.
+    """
+    email = serializers.CharField(required=False)
+    course_run_key = serializers.CharField(required=False)
+    license_uuid = serializers.CharField(required=False)
+
+    def create(self, validated_data):
+        return validated_data
+
+    def validate(self, data):  # pylint: disable=arguments-differ
+        missing_fields = []
+        for key in self.fields.keys():
+            if not data.get(key):
+                missing_fields.append(key)
+
+        if missing_fields:
+            raise serializers.ValidationError('Found missing licenses info field(s): {}.'.format(missing_fields))
+
+        return data
+
+
 # pylint: disable=abstract-method
 class EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(serializers.Serializer):
     """
     Serializes a licenses info field for bulk enrollment requests.
     """
-    licenses_info = serializers.ListField(
-        child=serializers.DictField(
-            child=serializers.CharField(required=False),
-        ),
-        required=False,
-    )
+    licenses_info = LicensesInfoSerializer(many=True, required=False)
```

Comment:
is using required=True here better than checking validation down below like you added? or did I miss sth? not sure if we get the same result .... but yours certainly have a custom message ... so worth checking which is better

Reply:
I was basing the pattern off of the existing single course bulk enrollment endpoint's serializer. I believe you're right in that it was done this way to handle customer messages as well as possibly the issue that Adrienne was investigating on the original PR I had for bulk enrollment. Do you have a preference here one way or the other?

Reply:
Nah I would not change it now. If the behavior you want is being met let's leave it. But semantically required=True would be nice. I just don't know enough right now to recommend setting it to true since you have a custom message. It seems that there is another way to get custom messages (https://www.django-rest-framework.org/api-guide/fields/#error_messages) which we can investigate if we want